### ISSUE TITLE
Rosetta results with error for any AMD64 binary

### ISSUE THREAD

**Issue by pszpetkowski** [utmapp/UTM#5567]
**Describe the issue**  
I have installed ARM64 Debian Trixie guest with virtualisation and Rosetta support enabled and then followed the documentation guide on how to setup Rosetta for Debian.

It seemed like everything is okay and I managed to install 7z (p7zip-full:amd64), but trying to run 7z (or any other AMD64 binary as it turned out) results in an error:
```
rosetta error: unhandled auxillary vector type 28
 Trace/breakpoint trap

```

Rosetta seems to work fine for macOS - I'm using quite a few Intel binaries. 

**Configuration**  
* UTM Version: 4.3.5 (87)
* macOS Version: 13.5
* Mac Chip (Intel, M1, ...): M1 Pro

**Crash log**  
If the app crashed, you need a crash log. To get your crash log, open Console.app, go to `Crash Reports`, and find the latest entry for either UTM, QEMU, QEMUHelper, or qemu-\*. Right click and choose `Reveal in Finder`. Attach the report here.

**Debug log**  
For all issues related to running a VM, _including_ crashes, you should attach a debug log. (This is unavailable for macOS-on-macOS VMs. Attach an excerpt of your system log instead.)
To get the Debug log: open UTM, and open the settings for the VM you wish to launch. Near the top of the `QEMU` page is `Debug Log`. Turn it on and save the VM. After you experience the issue, open the VM settings again and select `Export Log...` and attach it here.

**Upload VM**  
If your issue is related to a specific VM, please upload the config.plist from your VM's .utm directory. To get this, right-click the VM in UTM and choose "Show in Finder". Then right-click the selected file in Finder and choose "Show Package Contents". The config.plist file is now visible. Right-click it and choose "Compress". Attach the resulting config.plist.zip file here.
You can upload the entire .utm if needed, but note this includes your VM's drive image and may contain personal data. Since Github has an attachment size limit, you may want to upload to another service such as Google Drive. Link it here.

**Comment by pszpetkowski** [comment]
Curiously, it worked on Ubuntu. Maybe I'm missing something on my Debian VM?

**Comment by osy** [comment]
This is an issue only Apple can resolve. I would recommend reporting it here: https://www.apple.com/feedback/

**Comment by faxioman** [comment]
In my Debian VM, which I use as a Docker machine, running 'apt-get upgrade' solved the issue.

**Comment by faxioman** [comment]
I had the same issue after updating macOS to version 13.5.1. I had to modify the command to install Rosetta.

```
/usr/sbin/update-binfmts --install rosetta /media/rosetta/rosetta  \
    --magic "\x7fELF\x02\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\x3e\x00" \
    --mask "\xff\xff\xff\xff\xff\xfe\xfe\x00\xff\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff\xff" \
    --credentials yes --preserve yes --fix-binary yes
```

Note the `--preserve` parameter to `yes`.

**Comment by loopbaq** [comment]
I've got the same error when trying to run amd64 containers from Parallels in a Kali Linux ARM machine.
It actually works with Rosetta and Ubuntu but for some reason not with Kali Linux.
Has anyone solved this error?
Thank you.

**Comment by mediouni-m** [comment]
Hello,

This error is due to to two additional auxiliary vectors added in https://github.com/torvalds/linux/commit/317c8194e6aeb8b3b573ad139fc2a0635856498e. As a short-term workaround, the best option is to stay on Linux kernel versions older than 6.3 or set CONFIG_RSEQ=n when compiling your kernel.

I'll take care of reporting this ticket to Apple if I get some spare time to do so, especially as it's behaviour that constrains future evolution.

**Comment by EETagent** [comment]
I've had the issue open since 17 July, they finally found the bug and promised a fix, hopefully it will be in macOS 14 RC

![image](https://github.com/utmapp/UTM/assets/20557318/3e87775b-a773-4ad0-bc7f-b4c41649a171)

**Comment by EETagent** [comment]
They fixed it in  23B5046f (14,1 beta)

**Comment by hasan4791** [comment]
It's working fine now on 14.1
